evaluator/perception_evaluator/src/utils/marker_utils.cpp

```diff
@@ -0,0 +1,198 @@
+using tier4_autoware_utils::createMarkerScale;
+using tier4_autoware_utils::createPoint;
+using visualization_msgs::msg::Marker;
+
+void addFootprintMarker(
+  visualization_msgs::msg::Marker & marker, const geometry_msgs::msg::Pose & pose,
+  const vehicle_info_util::VehicleInfo & vehicle_info)
+{
+  const double half_width = vehicle_info.vehicle_width_m / 2.0;
+  const double base_to_front = vehicle_info.vehicle_length_m - vehicle_info.rear_overhang_m;
+  const double base_to_rear = vehicle_info.rear_overhang_m;
+
+  marker.points.push_back(
+    tier4_autoware_utils::calcOffsetPose(pose, base_to_front, -half_width, 0.0).position);
+  marker.points.push_back(
+    tier4_autoware_utils::calcOffsetPose(pose, base_to_front, half_width, 0.0).position);
+  marker.points.push_back(
+    tier4_autoware_utils::calcOffsetPose(pose, -base_to_rear, half_width, 0.0).position);
+  marker.points.push_back(
+    tier4_autoware_utils::calcOffsetPose(pose, -base_to_rear, -half_width, 0.0).position);
+  marker.points.push_back(marker.points.front());
+}
+
+MarkerArray createFootprintMarkerArray(
+  const Pose & base_link_pose, const vehicle_info_util::VehicleInfo vehicle_info,
+  const std::string && ns, const int32_t & id, const float & r, const float & g, const float & b)
+{
+  const auto current_time = rclcpp::Clock{RCL_ROS_TIME}.now();
+  MarkerArray msg;
+
+  Marker marker = createDefaultMarker(
+    "map", current_time, ns, id, Marker::LINE_STRIP, createMarkerScale(0.2, 0.2, 0.2),
+    createMarkerColor(r, g, b, 0.999));
+  marker.lifetime = rclcpp::Duration::from_seconds(1.5);
+
+  MarkerArray marker_array;
+
+  addFootprintMarker(marker, base_link_pose, vehicle_info);
+  msg.markers.push_back(marker);
+  return msg;
+}
+
+MarkerArray createPointsMarkerArray(
+  const std::vector<Point> points, const std::string & ns, const int32_t id, const float r,
+  const float g, const float b)
+{
+  auto marker = createDefaultMarker(
+    "map", rclcpp::Clock{RCL_ROS_TIME}.now(), ns, id, Marker::LINE_STRIP,
+    createMarkerScale(0.05, 0.0, 0.0), createMarkerColor(r, g, b, 0.999));
+
+  for (const auto & point : points) {
+    marker.points.push_back(point);
+  }
+
+  MarkerArray msg;
+  msg.markers.push_back(marker);
+  return msg;
+}
+
+MarkerArray createPointsMarkerArray(
+  const std::vector<Pose> poses, const std::string & ns, const int32_t id, const float r,
+  const float g, const float b)
+{
+  auto marker = createDefaultMarker(
+    "map", rclcpp::Clock{RCL_ROS_TIME}.now(), ns, id, Marker::LINE_STRIP,
+    createMarkerScale(0.05, 0.0, 0.0), createMarkerColor(r, g, b, 0.999));
+
+  for (const auto & pose : poses) {
+    marker.points.push_back(pose.position);
+  }
+
+  MarkerArray msg;
+  msg.markers.push_back(marker);
+  return msg;
+}
+
+MarkerArray createDeviationLines(
+  const std::vector<Pose> poses1, const std::vector<Pose> poses2, const std::string & ns,
+  const int32_t id, const float r, const float g, const float b)
```

Comment:
Please fix build error before PR. This `id` is not used.

Reply:
oh, thanks. fixed build werror in https://github.com/autowarefoundation/autoware.universe/pull/6493/commits/1d3e81021128f96290def911cb91f1a544d9daf1